### PR TITLE
[TwigBundle] Added pre-render events

### PR DESCRIPTION
**Pull request by ghost**
| Q             | A
| ------------- | ---
| Branch?       | feature for 4.4, 5.0
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Tickets       | 
| License       | MIT
| Doc PR        | symfony/symfony-docs#12731... 

Enabling pre render events. This way we can add context just before rendering the template. Usecase for rendering partials.

### CODE REVIEW

File: src/Symfony/Bundle/TwigBundle/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ CHANGELOG
 5.0.0
 -----
 
+ * Add event dispatcher to the config, by default escaped to null. related to: https://github.com/twigphp/Twig/pull/3207
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/TwigBundle/CHANGELOG.md]: please remove the URL, we don't add these - the message should be self-explanatory

File: src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -203,4 +204,16 @@ private function addTwigFormatOptions(ArrayNodeDefinition $rootNode)
             ->end()
         ;
     }
+
+    private function addEventDispatcher(ArrayNodeDefinition $rootNode)
+    {
+        $rootNode
+            ->children()
+                ->variableNode('event_dispatcher')
+                    ->info('The event dispatcher')
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php]: not sure this is useful - can we add more info why this is worth configuring? if not, remove it?

File: src/Symfony/Bundle/TwigBundle/CHANGELOG.md (hunk 1 above)
Comment (patrickSkepp) [reply src/Symfony/Bundle/TwigBundle/CHANGELOG.md]: Change the message, I hope this description will make it much better ;)

File: src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php (hunk 2 above)
Comment (patrickSkepp) [reply src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php]: Also changed the message, is this what you have in mind?

**Comment by nicolas-grekas** [comment]
Do we need to allow injecting any service id here, or can we turn this config option into a boolean and inject the `event_dispatcher` service?

Also: why don't we always configure the event_dispatcher and remove the option instead?

**Comment by yceruto** [comment]
I failed to see where this option comes from, is it a new option that hasn't been released yet?

**Comment by patrickSkepp** [comment]
@nicolas-grekas If you have your own implementation of the event dispatcher interface, it would be really nice if people can use it too, that is why I would say inject the service.
Also not everybody would like to use the event dispatcher to fire an event, It would be nice if it is optional.

@yceruto it is related to: https://github.com/twigphp/Twig/pull/3207

**Comment by xabbuh** [comment]
closing here as the related PR in the Twig repository has been rejected

**Pull request closed without merging**